lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioInputHelper.java

```diff
@@ -113,12 +110,11 @@ private int doInternalCacheRead(long position, byte[] bytes, int offset, int len
         int remainingLength = length;
         while (remainingLength > 0) {
             int bytesReadFromCache = readPageFromCache(position, bytes, offset, remainingLength);
-            if (bytesReadFromCache == 0) {
+            // When dealing with concurrent access for a new file, CacheManager#put doesn't guarantee the page is fully written,
+            // and trying to access the page could return -1, so we retry reading data
```

Comment:
by "retry" do you mean the retry from cache or fallback to remote source ? i'm assuming it's the latter, would be good to clarify that

Reply:
Updated them